pieces/shared/menu-button/container.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  KeyboardEventHandler,
+  MouseEventHandler,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useReducer,
+  useRef,
+  useState,
+} from 'react';
+
+import { css } from '@emotion/react';
+
+import Trigger from './trigger';
+import Menu, { MenuHandle } from './menu';
+
+const containerStyles = css({
+  display: 'flex',
+  position: 'relative',
+});
+
+const useInitialRender = () => {
+  const isInitialRenderRef = useRef(true);
+  useEffect(() => {
+    isInitialRenderRef.current = false;
+  }, []);
+  return { isInitialRender: isInitialRenderRef.current };
+};
+
+export const MenuButton = ({ label, children }: { label: string; children: ReactNode }) => {
+  const [isOpen, toggleIsOpen] = useReducer((isOpen) => !isOpen, false);
+
+  const [initialFocus, setInitialFocus] = useState<'first' | 'last'>('first');
+
+  const menuRef = useRef<MenuHandle>(null);
+
+  const triggerRef = useRef<HTMLButtonElement>(null);
+
+  const shouldResetFocusRef = useRef(false);
+
+  useEffect(() => {
+    if (!isOpen && shouldResetFocusRef.current) {
```

Comment:
why is this needed?

Reply:
hopefully the changed code should make this clearer